### PR TITLE
[Melodic] Fix for 360° lidars.

### PR DESCRIPTION
**Pull request by SelvamArul**
I followed option 2 as you mention in #141 comments.
Please take a look at the changes, and lemme know if it looks good.

### CODE REVIEW

File: src/laser_utils.cpp [hunk 1]
```diff
@@ -101,6 +102,13 @@ karto::LaserRangeFinder* LaserAssistant::makeLaser(const double& mountingYaw)
   laser->SetMinimumAngle(scan_.angle_min);
   laser->SetMaximumAngle(scan_.angle_max);
   laser->SetAngularResolution(scan_.angle_increment);
+  
+  bool is_360_lidar = false;
+  if ( (std::fabs(scan_.angle_min + M_PI) < 1e-3) &&
```
Comment (SteveMacenski) [inline src/laser_utils.cpp:107]: huh, not the way I would have done this but probably cleaner. Nice.

File: src/laser_utils.cpp [hunk 2]
```diff
@@ -101,6 +102,13 @@ karto::LaserRangeFinder* LaserAssistant::makeLaser(const double& mountingYaw)
   laser->SetMinimumAngle(scan_.angle_min);
   laser->SetMaximumAngle(scan_.angle_max);
   laser->SetAngularResolution(scan_.angle_increment);
+  
+  bool is_360_lidar = false;
+  if ( (std::fabs(scan_.angle_min + M_PI) < 1e-3) && 
+    (std::fabs(scan_.angle_max - M_PI) < 1e-3))
+    is_360_lidar = true;
+
+  laser->SetIs360Laser(is_360_lidar);
```
Comment (SteveMacenski) [inline src/laser_utils.cpp:111]: You're setting a bool, but getting a double, that doesn't make much sense.

File: lib/karto_sdk/include/karto_sdk/Karto.h [hunk 3]
```diff
@@ -4338,9 +4362,10 @@ namespace karto
      */
     void Update()
     {
+      int laserResidual = !GetIs360Laser();
```
Comment (SteveMacenski) [inline lib/karto_sdk/include/karto_sdk/Karto.h]: You find if its 360 and give it a bool, the variable is implicitly stored as a bool, the `Get` is a double, and you're assigning it an int. This needs to be cleaned up.

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

**Comment by SteveMacenski** [comment]
Better option than the 2 I provided, but need to make sure the types are consistent and valid. If you store the "is360lidar" that is implicitly a boolean. You should assign the int for how many to add based on the boolean value. 

Basically: the `get` and `set` should be `bool`s, and you can use the `Get` to assign the `residual` as `0` or `1`.

**Comment by SelvamArul** [comment]
Ja, the return type of  `GetIs360Laser` should have been `kt_bool`. It was copy-paste error. I fixed it. Getting `is360lidar` as bool and assigning `laserResidual` (of type int) based on `is360lidar` is done as you suggested :)

**Comment by SteveMacenski** [comment]
@SelvamArul thanks for fixing the `kt_double`!

Still a (bug? not really a bug... confusing abstraction?) thing with the `int residual = !GetIs360Lidar()`.

I see logically that is sound, but its really not straight forward that you're converting the inverse of a boolean as an integer. It makes it hard to read and understand at first glance. Can you change to something like:

```
int residual = 1;
if (GetIs360Lidar()) 
{
  // residual is 0 by 360 lidar conventions
  residual = 0;
}

```

**Comment by SelvamArul** [comment]
I agree with bool to int conversion readability. Changed the code to make it more explicit as you suggested.

**Review by SteveMacenski — APPROVED** [review]
LGTM

Letting it sit for a hot minute while I cherry pick over to Eloquent and Dashing. If its not still merged next time you see it and you're willing, I'd appreciate it. There's alot on my plate this week.

**Comment by SteveMacenski** [comment]
Thanks for your contribution!

**Pull request merged**